nautobot/core/views/renderers.py

```diff
@@ -221,25 +228,30 @@ def get_context(self, data, accepted_media_type, renderer_context):
             elif view.action == "changelog":
                 table = self.construct_table(view, object=instance, content_type=content_type)
 
-        context = {
-            "content_type": content_type,
-            "form": form,
-            "filter_form": filter_form,
-            "dynamic_filter_form": self.get_dynamic_filter_form(view, request, filterset_class=view.filterset_class),
-            "search_form": search_form,
-            "filter_params": display_filter_params,
-            "object": instance,
-            "obj": instance,  # NOTE: This context key is deprecated in favor of `object`.
-            "obj_type": queryset.model._meta.verbose_name,  # NOTE: This context key is deprecated in favor of `verbose_name`.
-            "obj_type_plural": queryset.model._meta.verbose_name_plural,  # NOTE: This context key is deprecated in favor of `verbose_name_plural`.
-            "permissions": permissions,
-            "return_url": return_url,
-            "table": table if table is not None else data.get("table", None),
-            "table_config_form": TableConfigForm(table=table) if table else None,
-            "verbose_name": queryset.model._meta.verbose_name,
-            "verbose_name_plural": queryset.model._meta.verbose_name_plural,
-        }
-        if view.action == "retrieve":
+        context = self.get_template_context(data, renderer_context)
+        context.update(
+            {
+                "content_type": content_type,
+                "form": form,
+                "filter_form": filter_form,
+                "dynamic_filter_form": self.get_dynamic_filter_form(
+                    view, request, filterset_class=view.filterset_class
+                ),
+                "search_form": search_form,
+                "filter_params": display_filter_params,
+                "object": instance,
+                "obj": instance,  # NOTE: This context key is deprecated in favor of `object`.
+                "obj_type": queryset.model._meta.verbose_name,  # NOTE: This context key is deprecated in favor of `verbose_name`.
+                "obj_type_plural": queryset.model._meta.verbose_name_plural,  # NOTE: This context key is deprecated in favor of `verbose_name_plural`.
+                "permissions": permissions,
+                "return_url": return_url,
+                "table": table if table is not None else data.get("table", None),
+                "table_config_form": TableConfigForm(table=table) if table else None,
+                "verbose_name": queryset.model._meta.verbose_name,
+                "verbose_name_plural": queryset.model._meta.verbose_name_plural,
+            }
+        )
+        if view.detail:
```

Comment:
This probably needs further refinement but I made the change to support our documented extra actions [here](https://docs.nautobot.com/projects/core/en/stable/development/apps/api/views/nautobotuiviewset/#adding-custom-views-to-nautobotuiviewset-nautobotuiviewsetrouter).

It seems wrong to only call the viewset's `get_extra_context` for the retrieve view. That should be common to all views in the viewset.

The `common_detail_view_context` does look specific to the retrieve view so perhaps we keep that behavior and rename the function to `common_retrieve_view_context` to avoid confusing DRF's [`detail=True (boolean indicating if the current action is configured for a list or detail view.)`](https://www.django-rest-framework.org/api-guide/viewsets/#introspecting-viewset-actions) for views. I'd actually prefer to move a lot of this logic into the UIViewSet mixins, but not in this goliath of a PR.